### ISSUE TITLE
Easy Config-related cleanups

### ISSUE THREAD

**Issue by dmlloyd** [quarkusio/quarkus#5388]
**Description**
After #5387, there are several configuration constructs that can be cleaned up in core and extensions.

- [x] Any `@ConfigItem` which has a primitive type like `int` or `boolean` that has an explicit default value of `0` or `false` can have the default value removed.
- [x] Any `@ConfigItem` with a primitive wrapper type like `Integer` or `Boolean` with an explicit default value of `0` or `false` should be changed to use the primitive type instead and remove the default value.
- [ ] Any `@ConfigItem` of type `Optional<Integer>` can be changed to `OptionalInt`.
- [ ] Any `@ConfigItem` which is a `List` type that is always converted to an array using `toArray` can be changed to just be an array type instead.

**Comment by ShubhamRwt** [comment]
@dmlloyd  Is this issue fixed now? I was planning to work on it.

**Comment by dmlloyd** [comment]
@ShubhamRwt - the issue is mostly done but the PR (#5489) is out of date and needs a rebase.  The original contributor gave up in frustration (this was really our fault) but if you want to start with that commit, rebase it and bring it up to date (and add yourself as `Co-authored-by`) and submit a new PR, go ahead and do so.  Thanks!

**Comment by ShubhamRwt** [comment]
Ok @dmlloyd. I will rebase it and send a new PR.

**Comment by machi1990** [comment]
Thanks @ShubhamRwt

**Comment by ShubhamRwt** [comment]
Hello @dmlloyd @machi1990 . The previous guy has made some unwanted changes due to which the build is failing. Should I create a new Pull request and make the only required changes?

**Comment by dmlloyd** [comment]
Go ahead.  I'd recommend fixing just one problem per PR at a time; that will make it more likely to get merged quickly and hopefully avoid the troubles that the previous contributor encountered.

**Comment by ShubhamRwt** [comment]
Yes Sure. That would be great

**Comment by ShubhamRwt** [comment]
@machi1990 I went through the code base but seems like the second part is already resolved. Actually I was stuck with job work lately thats why this didn't got time for the rest over part but I will get it done and sorry for the delay.

**Comment by ShubhamRwt** [comment]
I will be proceeding with the 3rd pointer mentioned above tomorrow.

**Comment by machi1990** [comment]
@ShubhamRwt thanks for the update. No rush, and take your time.

**Comment by toniperezfer** [comment]
Hi. I would like to contribute to the project. It seems like there are two unresolved problems yet, right?

**Comment by adrianfiedler** [comment]
Hi @dmlloyd, @machi1990,
for third task I prepared a PR, could you please check?
https://github.com/quarkusio/quarkus/pull/10440

**Comment by machi1990** [comment]
> Hi @dmlloyd, @machi1990,
> for third task I prepared a PR, could you please check?
> #10440

Thanks for the PR @adrianfiedler I'll have a look later today.
